owncloudApp/src/main/java/com/owncloud/android/ui/activity/ReceiveExternalFilesActivity.java

```diff
@@ -346,6 +405,33 @@ public void onClick(DialogInterface dialog, int which) {
                     setAccount(mAccountManager.getAccountsByType(MainApp.Companion.getAccountType())[which]);
                     onAccountSet(mAccountWasRestored);
                     dialog.dismiss();
+                    PreferenceManager.setLastUploadPath("/", this);
+                    root = getStorageManager().getRootPersonalFolder();
+                    if (root.getSpaceId() == null) {
+                        showFilesview();
+                        showRetainerFragment();
+                        updateDirectoryList();
+                        if (mParents.size() == 1) {
+                            updateToolbar(getString(R.string.uploader_top_message));
+                        }
+
+                    } else {
+                        mListView = findViewById(android.R.id.list);
+                        fragmentContainer = findViewById(R.id.fragment_container);
+                        mListView.setVisibility(View.GONE);
+                        fragmentContainer.setVisibility(View.VISIBLE);
+                        initAndShowListOfSpaces();
+                        updateToolbar(getString(R.string.app_name));
+                        getSupportFragmentManager().setFragmentResultListener(SpacesListFragment.REQUEST_KEY_CLICK_SPACE, this, (requestKey,
+                                                                                                                                 bundle) -> {
```

Comment:
I see this is the same as in `onCreate`. Doesn't `onCreate` method executes after choosing the account? Or do we need both because the code in `onCreate` is useful only for the case where we only have 1 account and this inside the `DIALOG_MULTIPLE_ACCOUNT` `onClick` only for the case where we have more than 1 account? Please enlighten me 💡

Reply:
Exactly, depending on the flow, the code differs a little, since when adding multi-account the functionality changes a little.